frontend/src/Admin/actions.js

```diff
@@ -0,0 +1,97 @@
+import Promise from 'bluebird'
+
+import ShowQuery from './queries/show.graphql'
+import ShowsQuery from './queries/shows.graphql'
+import JudgesQuery from './queries/judges.graphql'
+import JudgesForShowQuery from './queries/judgesForShow.graphql'
+
+export const LOADING_DATA = 'LOADING_DATA'
+export const FETCH_SHOW = 'FETCH_SHOW'
+export const FETCH_SHOWS = 'FETCH_SHOWS'
+export const FETCH_JUDGES = 'FETCH_JUDGES'
+export const FETCH_JUDGES_FOR_SHOW = 'FETCH_JUDGES_FOR_SHOW'
+export const FETCH_JUDGES_BY_ASSIGNMENT_FOR_SHOW = 'FETCH_JUDGES_BY_ASSIGNMENT_FOR_SHOW'
+export const ASSIGN_JUDGES_TO_SHOW = 'ASSIGN_JUDGES_TO_SHOW'
+export const REMOVE_JUDGES_FROM_SHOW = 'REMOVE_JUDGES_FROM_SHOW'
+
+export const fetchShow = (showId) => (dispatch, getState, client) => {
+  return client.query({
+    query: ShowQuery,
+    variables: {
+      id: showId
+    }
+  })
+    .then(({data: {show}}) => dispatch({type: FETCH_SHOW, payload: show}))
+    .catch(console.error) // TODO: Handle the error
+}
+
+export const fetchShows = () => (dispatch, getState, client) => {
+  return client.query({query: ShowsQuery})
+    .then(({data: {shows}}) => dispatch({type: FETCH_SHOWS, payload: shows}))
+    .catch(console.error) // TODO: Handle the error
+}
+
+export const fetchJudges = () => (dispatch, getState, client) => {
+  return client.query({query: JudgesQuery})
+    .then(({data: {judges}}) => dispatch({type: FETCH_JUDGES, payload: judges}))
+    .catch(console.error) // TODO: Handle the error
+}
+
+export const fetchJudgesForShow = (showId) => (dispatch, getState, client) => {
+  return client.query({
+    query: JudgesForShowQuery,
+    variables: {
+      id: showId
+    }
+  })
+    .then(({data: {show}}) => dispatch({type: FETCH_JUDGES_FOR_SHOW, payload: show}))
+    .catch(console.error) // TODO: Handle the error
+}
+
+export const fetchJudgesByAssignmentForShow = (showId) => (dispatch, getState, client) => {
+  return Promise
+    .all([
+      client.query({
+        query: JudgesForShowQuery,
+        variables: {
+          id: showId
+        }
+      }),
+      dispatch(fetchShows()),
+      dispatch(fetchJudges())
+    ])
+    .spread(({data: {show}, loading}) => {
```

Comment:
Based on the way we normalize our state and how Apollo caches data, if someone goes to the Judge assignment page directly, we need to load in Shows, Judges, and the Judge assignments between shows, which is why I dispatch 3 actions here. 

To answer 1, I need to look into the React lifecycle w/ Apollo more because uninitialized data (eg. `null`, `undefined`) seems to flow through `render` a few times before actual data. Now that I re-look at this, I don't even know if `loading` will be false based on using `Promise.all`. I'll have to look into Apollo's `loading` (and `error`) props more because their docs only have examples using their `graphql` higher-order function, not w/ using Redux.

To answer 2, pretty sure it does. See `mapDispatchToProps` in one of the containers below. Has to do with how `fetchShows()` is a function which returns a function.